### DISCUSSION TITLE
BERTopic for multi-modal data

### DISCUSSION THREAD

**Discussion by david-schweizer** [MaartenGr/BERTopic#1234] [Q&A]
Hey community,

does anyone know if it is possible to run BERTopic on multi-modal data (text and images)? In particular, I have Twitter data, i.e., sometimes only text and sometimes text and image.

I have seen in the documentation that there is the possibility of including captions for images, but I wonder if I could use the text of the Tweet instead.

Also, do you have any idea how I could deal with the issue that my data consists sometimes only of text and sometimes of text and images? 

Best,
David

**Comment by MaartenGr** [comment]
When dealing with multi-modal data, [this](https://maartengr.github.io/BERTopic/getting_started/tips_and_tricks/tips_and_tricks.html#multi-modal-data) is a good place to start with BERTopic. More specifically, since you always have at least text, I would advise passing the text as the main thing to perform topic modeling on. If you want to add images to that, it might make sense to use an embedding technique that can create embeddings from both images and text. Then, you could average those embeddings for the instances where you have both text and images. 

In practice, it would look something like this:

```python
topic_model = BERTopic()
topics, probs = topic_model.fit_transform(tweets, embeddings)
```

where `tweets` are your tweets and `embeddings` are generated from:

```python
from sentence_transformers import SentenceTransformer
model = SentenceTransformer('clip-ViT-B-32')
doc_embeddings = model.encode(tweets)
image_embeddings = model.encode(images)
```

Then, you need to merge `doc_embeddings` and `image_embeddings` for the tweets that also have images. 

If you have images for each tweet, then the [upcoming multi-modal update](https://github.com/MaartenGr/BERTopic/pull/1232) might be better suited.
